### PR TITLE
Add suricata update

### PR DESCRIPTION
**Pull request by masami256**
# Purpose of pull request

This PR added suricata-update package to download/update suricata's rule files. It also added  packages which suricata-update package depend.

# Test
### setup 

- add following lines in conf/local.conf

```
IMAGE_ROOTFS_EXTRA_SPACE = "5242880"
IMAGE_INSTALL_append = " suricata suricata-update"
```

- build core-image-minimal
- run qemuarm64(suricata-update doesn't work with default runqemu's memory size)

```
$ runqemu nographic qemuarm64 qemuparams= -m 2048
```
- remove unnecessary rules(such as smb) from /etc/suricata/suricata.yaml 

- Run sucicate-update command

```
# suricata-update 
```

If command success, exit code shoud be zero, otherwise non-zero value returns.

## Test result

```
root@qemuarm64:~# suricata-update
[  378.374223] random: python3: uninitialized urandom read (24 bytes read)
[  382.120788] random: crng init done
20/1/2021 -- 04:08:53 - <Info> -- Using data-directory /var/lib/suricata.
20/1/2021 -- 04:08:53 - <Info> -- Using Suricata configuration /etc/suricata/suricata.yaml
20/1/2021 -- 04:08:53 - <Info> -- Using /etc/suricata/rules for Suricata provided rules.
20/1/2021 -- 04:08:53 - <Info> -- Found Suricata version 4.1.2 at /usr/bin/suricata.
20/1/2021 -- 04:08:53 - <Info> -- Loading /etc/suricata/suricata.yaml
20/1/2021 -- 04:08:53 - <Info> -- Disabling rules with proto krb5
20/1/2021 -- 04:08:53 - <Info> -- Disabling rules with proto ikev2
20/1/2021 -- 04:08:53 - <Info> -- Disabling rules with proto dcerpc
20/1/2021 -- 04:08:53 - <Info> -- Disabling rules with proto imap
20/1/2021 -- 04:08:53 - <Info> -- Disabling rules with proto msn
20/1/2021 -- 04:08:53 - <Info> -- Disabling rules with proto smb
20/1/2021 -- 04:08:53 - <Info> -- Disabling rules with proto nfs
20/1/2021 -- 04:08:53 - <Info> -- Disabling rules with proto tftp
20/1/2021 -- 04:08:53 - <Info> -- Disabling rules with proto modbus
20/1/2021 -- 04:08:53 - <Info> -- Disabling rules with proto dnp3
20/1/2021 -- 04:08:53 - <Info> -- Disabling rules with proto enip
20/1/2021 -- 04:08:53 - <Info> -- Disabling rules with proto ntp
20/1/2021 -- 04:08:53 - <Info> -- Disabling rules with proto dhcp
20/1/2021 -- 04:08:53 - <Info> -- No sources configured, will use Emerging Threats Open
20/1/2021 -- 04:08:53 - <Info> -- Fetching https://rules.emergingthreats.net/open/suricata-4.1.2/emerging.rules.tar.gz.
20/1/2021 -- 04:08:54 - <Error> -- Failed to fetch https://rules.emergingthreats.net/open/suricata-4.1.2/emerging.rules.tar.gz: <urlopen error [Errno -3] Temporary failure in name resolution>
20/1/2021 -- 04:08:54 - <Info> -- Loading distribution rule file /etc/suricata/rules/app-layer-events.rules
20/1/2021 -- 04:08:54 - <Info> -- Loading distribution rule file /etc/suricata/rules/decoder-events.rules
20/1/2021 -- 04:08:54 - <Info> -- Loading distribution rule file /etc/suricata/rules/dnp3-events.rules
20/1/2021 -- 04:08:54 - <Info> -- Loading distribution rule file /etc/suricata/rules/dns-events.rules
20/1/2021 -- 04:08:54 - <Info> -- Loading distribution rule file /etc/suricata/rules/files.rules
20/1/2021 -- 04:08:54 - <Info> -- Loading distribution rule file /etc/suricata/rules/http-events.rules
20/1/2021 -- 04:08:54 - <Info> -- Loading distribution rule file /etc/suricata/rules/ipsec-events.rules
20/1/2021 -- 04:08:54 - <Info> -- Loading distribution rule file /etc/suricata/rules/kerberos-events.rules
20/1/2021 -- 04:08:54 - <Info> -- Loading distribution rule file /etc/suricata/rules/modbus-events.rules
20/1/2021 -- 04:08:54 - <Info> -- Loading distribution rule file /etc/suricata/rules/nfs-events.rules
20/1/2021 -- 04:08:54 - <Info> -- Loading distribution rule file /etc/suricata/rules/ntp-events.rules
20/1/2021 -- 04:08:54 - <Info> -- Loading distribution rule file /etc/suricata/rules/smb-events.rules
20/1/2021 -- 04:08:54 - <Info> -- Loading distribution rule file /etc/suricata/rules/smtp-events.rules
20/1/2021 -- 04:08:54 - <Info> -- Loading distribution rule file /etc/suricata/rules/stream-events.rules
20/1/2021 -- 04:08:54 - <Info> -- Loading distribution rule file /etc/suricata/rules/tls-events.rules
20/1/2021 -- 04:08:55 - <Info> -- Loaded 324 rules.
20/1/2021 -- 04:08:55 - <Info> -- Disabled 39 rules.
20/1/2021 -- 04:08:55 - <Info> -- Enabled 0 rules.
20/1/2021 -- 04:08:55 - <Info> -- Modified 0 rules.
20/1/2021 -- 04:08:55 - <Info> -- Dropped 0 rules.
20/1/2021 -- 04:08:55 - <Info> -- Enabled 0 rules for flowbit dependencies.
20/1/2021 -- 04:08:55 - <Info> -- Creating directory /var/lib/suricata/rules.
20/1/2021 -- 04:08:55 - <Info> -- Backing up current rules.
20/1/2021 -- 04:08:55 - <Info> -- Writing rules to /var/lib/suricata/rules/suricata.rules: total: 324; enabled: 244; added: 324; removed 0; modified: 0
20/1/2021 -- 04:08:55 - <Info> -- Testing with suricata -T.
20/1/2021 -- 04:08:56 - <Info> -- Done.
root@qemuarm64:~# echo $?
0
```

### CODE REVIEW

File: recipes-debian/cython/python3-cython_debian.bb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# base recipe: recipes-python/python-cython-native/python3-cython-native_0.29.21.bb
+# base branch: master
+# base commit: 5871650434275c463e59e53541cc9a7bdcab9511
+# https://github.com/priv-kweihmann/meta-sca/blob/master/recipes-python/python-cython-native/python3-cython-native_0.29.21.bb
```
Comment (hiraku-wfs) [inline recipes-debian/cython/python3-cython_debian.bb]: Why don't you refer the poky/oe?
https://layers.openembedded.org/layerindex/recipe/121388/

File: recipes-debian/pyyaml/python3-pyyaml_debian.bb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# base recipe: recipes-devtools/python/python3-pyyaml_5.3.1.bb
```
Comment (hiraku-wfs) [inline recipes-debian/pyyaml/python3-pyyaml_debian.bb]: Please prepend "meta-python/"

File: recipes-debian/scons/python3-scons-native_debian.bb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# base recipe: /recipes-devtools/python/python3-scons-native_3.1.2.bb
```
Comment (hiraku-wfs) [inline recipes-debian/scons/python3-scons-native_debian.bb]: Please prepend "meta/"

File: recipes-debian/scons/python3-scons-native_debian.bb [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+# base branch: master
+# base commit: 	44b46ae6c65448901477def57a45d5a4ecc2d246
```
Comment (hiraku-wfs) [inline recipes-debian/scons/python3-scons-native_debian.bb]: There is an unnecessary tab here.

File: recipes-debian/suricata-update/suricata-update_debian.bb [hunk 5]
```diff
@@ -0,0 +1,18 @@
+# base recipe: recipes-ids/suricata/python3-suricata-update_1.1.1.bb
```
Comment (hiraku-wfs) [inline recipes-debian/suricata-update/suricata-update_debian.bb]: Please prepend "meta-security/".

File: recipes-debian/pyyaml/python3-pyyaml_debian.bb (hunk 2 above)
Comment (masami256) [reply recipes-debian/pyyaml/python3-pyyaml_debian.bb]: fixed.

File: recipes-debian/scons/python3-scons-native_debian.bb (hunk 3 above)
Comment (masami256) [reply recipes-debian/scons/python3-scons-native_debian.bb]: fixed.

File: recipes-debian/scons/python3-scons-native_debian.bb (hunk 4 above)
Comment (masami256) [reply recipes-debian/scons/python3-scons-native_debian.bb]: fixed.

File: recipes-debian/suricata-update/suricata-update_debian.bb (hunk 5 above)
Comment (masami256) [reply recipes-debian/suricata-update/suricata-update_debian.bb]: fixed.

File: recipes-debian/cython/python3-cython_debian.bb (hunk 1 above)
Comment (masami256) [reply recipes-debian/cython/python3-cython_debian.bb]: I couldn't find this repo. thanks.
I changed base recipe and added pyparsing package that is required by cython.

**Comment by masami256** [comment]
@hiraku-wfs I pushed fixed version. PTAL.

File: recipes-debian/cython/python3-cython_debian.bb [hunk 6]
```diff
@@ -0,0 +1,29 @@
+# base recipe: meta/recipes-devtools/python/python-cython.inc
```
Comment (hiraku-wfs) [inline recipes-debian/cython/python3-cython_debian.bb]: You mean python3-cython.bb ?

Comment (masami256) [reply recipes-debian/cython/python3-cython_debian.bb]: @hiraku-wfs  oops, I fixed this one.

**Comment by hiraku-wfs** [comment]
Thanks!

**Pull request merged**